docs: :memo: update MIT license post (#164)
seedcase-project/decisions: why-mit-license/index.qmd

---
title: "Why an MIT License"
description: |
  Open and permissive licensing of copyrighted content is a way to increase use of a software product,

>   ease collaboration and re-use, and sharing intellectual ideas with others. This post contains our

Reviewer: ```suggestion
  ease collaboration and re-use, and share intellectual ideas with others. This post contains our
```

  reasons for choosing the MIT License, which is a very permissive copyright license.
date: "2023-04-01"

> date-modified: "2025-02-20"

Reviewer: Why not `last-modified` anymore?

Author: Doesn't work as I expected. It shows the date the HTML file was last generated... which is everything we rebuild the website after a PR is merged.

Given our drivers, we focus on permissive licenses, which are:

-   [MIT License](https://opensource.org/licenses/MIT)
-   [Apache License 2.0](https://opensource.org/licenses/Apache-2.0)
-   [BSD 3-Clause
    License](https://opensource.org/license/bsd-3-clause/))

### MIT License

Of the three licenses, the MIT License is the most permissive and with
the least restrictions.

::::: columns
::: column
#### Benefits

-   Very popular and widely used with the open source community, for
    example, most of Python, R, and JavaScript packages are licensed
    under the MIT License.
-   Very easy to understand and implement, with minimal legal jargon. It
    has a total of some 200 words, so it is very easy to read.
-   Allows for commercial use and distribution, with no requirement to
    release the source code of any derived works.
-   Requires nearly no legal enforcement, as it is so permissive.
-   Includes the software's documentation as part of it's license
    coverage.
:::

::: column
#### Drawbacks

-   The only major drawback is the same thing as it's strength: It is so
    permissive that it allows for anyone to use the software in any way

>     they see fit, including in proprietary software. This can be seen as

Reviewer: ```suggestion
    they see fit, including proprietary software. This can be seen as
```

    a drawback if you want to ensure that your software remains open
    source.
:::
:::::

### Apache License 2.0

The Apache License 2.0 is very similar to the MIT License, but with a
few more restrictions.

::::: columns
::: column
#### Benefits

-   Quite popular in the open source community.
-   Includes a section on patents, which can be useful if you are a
    company with patents.
:::

::: column
#### Drawbacks

-   Is a bit more complex than the MIT License, with more legal jargon.
-   Requires some legal enforcement to ensure compliance.
-   The patent section is a bit difficult to understand as a developer,

>     so presumably is difficult for users.

Reviewer: ```suggestion
    so presumably it is also difficult for users to understand.
```

-   Doesn't include the documentation as part of the license coverage.
:::
:::::

### BSD 3-Clause License

The BSD 3-Clause License is similar to both the other licenses, but with
less restrictions than the Apache License 2.0 and slightly more than the
MIT License. There is also the BSD 2-Clause License, but it is not as
commonly used and, while nearly identical to the MIT License, is less

> specific in it's language. In simple terms, it is a worse version of the

Reviewer: ```suggestion
specific in it's language. In simple terms, the BSD 2-Clause License is a worse version of the
```

> MIT License. So we don't covered it here.

Reviewer: ```suggestion
MIT License, so we don't cover it here.
```

::::: columns
::: column
#### Benefits

-   Is nearly identical to the MIT License, but has a clause to indicate
    not to use the name of the copyright holder to endorse or promote
    products derived from the software.
:::

::: column
#### Drawbacks

-   Only covers the software and it's binary format, not the
    documentation.

> -   The least commonly used.

Reviewer: ```suggestion
-   The least commonly used of the licenses considered here.
```
